dask/blockwise.py

```diff
@@ -17,6 +22,20 @@
 )
 
 
+def blockwise_generate_chunk(idx, chunks):
+    """Generate a local dask.array chunk"""
+    ndim = len(chunks)
+    return tuple(chunks[i][idx[i]] for i in range(ndim))
```

Comment:
```suggestion
    return tuple(chunk[i] for i, chunk in zip(idx, chunks))
```
unless idx is a a dict, in which case ``chunk[idx[i]] for i, chunk in enumerate(chunks)`` should be used. Please use type annotations.